### DISCUSSION TITLE
virtual modifier for the Id in the 'Entity' class

### DISCUSSION THREAD

**Discussion by ElMehdiSmiri** [onebeyond/monaco#40] [Q&A] [answered]
@CesarD  Why do we need the `virtual` modifier for the `Id` in the `Entity` class?

```
public abstract class Entity : IEntity
{
	protected Entity()
	{
	}

	protected Entity(Guid id) : this()
	{
		Id = id;
	}

	/// <summary>
	/// The identifier of the entity
	/// </summary>
	public virtual Guid Id { get; }
.....
```

**Comment by Gonzo345** [comment, marked as answer]
Wouldn't you need it to perform the tests? 🤔

**Reply by CesarD** [reply]
Yup, pretty much for that. Mainly to let anyone override the property behavior in a derived class, but mostly useful for overriding its value during mocking for testing.
